### PR TITLE
v1.6 backports 2020-06-30

### PR DESCRIPTION
**Pull request by aanm**
* #12106 -- Use right CCNP validation (@aanm)

Once this PR is merged, you can update the PR labels via:
```upstream-prs
$ for pr in 12106; do contrib/backporting/set-labels.py $pr done 1.6; done
```

### CODE REVIEW

**Comment by aanm** [comment]
test-backport-1.6

File: cilium/cmd/preflight_k8s_valid_cnp.go [hunk 1]
```diff
@@ -69,28 +71,38 @@ func validateCNPs() error {
...
-	ctx, initCancel := context.WithTimeout(context.Background(), validateK8sPoliciesTimeout)
-	defer initCancel()
-	cnpErr := validateNPResources(ctx, "ciliumnetworkpolicies", "CiliumNetworkPolicy")
+	restConfig, err := k8s.CreateConfig()
+	if err != nil {
+		return fmt.Errorf("Unable to create rest configuration for k8s CRD: %w", err)
+	}
+	apiExtensionsClient, err := apiextensionsclient.NewForConfig(restConfig)
+	if err != nil {
+		return fmt.Errorf("Unable to create API extensions clientset for k8s CRD: %w", err)
```
Comment (joestringer) [inline cilium/cmd/preflight_k8s_valid_cnp.go]: Travis reports:

```
cilium/cmd/preflight_k8s_valid_cnp.go:76:10: Errorf format %w has unknown verb w

cilium/cmd/preflight_k8s_valid_cnp.go:80:10: Errorf format %w has unknown verb w
```

This seems to be due to `%w` being a Go 1.13 feature. I think we just need to do:

```suggestion
		return fmt.Errorf("Unable to create rest configuration for k8s CRD: %s", err)
	}
	apiExtensionsClient, err := apiextensionsclient.NewForConfig(restConfig)
	if err != nil {
		return fmt.Errorf("Unable to create API extensions clientset for k8s CRD: %s", err)
```

**Review by joestringer — CHANGES_REQUESTED** [review]

**Comment by joestringer** [comment]
Looks like the only failures were the unit tests, so I fixed it up & will re-push.

**Comment by joestringer** [comment]
test-backport-1.6

**Review by joestringer — APPROVED** [review]

**Comment by joestringer** [comment]
@pchaigno quick reminder to run the command in the commit message to update labels after merging :-) I did it this time.

**Comment by pchaigno** [comment]
Ah, I thought the backporter was in charge of that. But yeah makes sense, since backporter could be offline and we want to do that as quickly as possible.

**Pull request merged**